tests/unit_tests/test_data_structure.py

```diff
@@ -0,0 +1,179 @@
+    ),
+    (
+        ("pass", "notchecked"),
+        {"fail": 442, "pass": 0, "unknown_error": 0, "other": 0, "sum_of_rules": 442}
+    ),
+    (
+        ("notchecked"),
+        {"fail": 442, "pass": 191, "unknown_error": 0, "other": 0, "sum_of_rules": 633}
+    ),
+    (
+        ("error", "unknown"),
+        {"fail": 442, "pass": 191, "unknown_error": 0, "other": 69, "sum_of_rules": 702}
+    ),
+    (
+        ("notselected", "notapplicable"),
+        {"fail": 442, "pass": 191, "unknown_error": 0, "other": 69, "sum_of_rules": 702}
+    ),
+])
+def test_report_rule_results_stats(to_remove, result):
+    report = remove_all_rules_by_result(get_report(), to_remove)
+    if "raise" not in result:
+        rule_results_stats = report.get_rule_results_stats()
+        for key in result:
+            assert result[key] == rule_results_stats[key]
+    else:
+        with pytest.raises(MissingProcessableRules):
+            assert report.get_rule_results_stats()
+
+
+@pytest.mark.parametrize("to_remove, result", [
+    (
+        (),
+        {"low": 33, "medium": 351, "high": 25, "unknown": 33, "sum_of_filed_rules": 442}
+    ),
+    (
+        ("low", "medium", "high", "unknown"),
+        {"raise": True}
+    ),
+    (
+        ("fail"),
+        {"raise": True}
+    ),
+    (
+        ("low"),
+        {"low": 0, "medium": 351, "high": 25, "unknown": 33, "sum_of_filed_rules": 409}
+    ),
+    (
+        ("medium"),
+        {"low": 33, "medium": 0, "high": 25, "unknown": 33, "sum_of_filed_rules": 91}
+    ),
+    (
+        ("high"),
+        {"low": 33, "medium": 351, "high": 0, "unknown": 33, "sum_of_filed_rules": 417}
+    ),
+    (
+        ("unknown"),
+        {"low": 33, "medium": 351, "high": 25, "unknown": 0, "sum_of_filed_rules": 409}
+    ),
+    (
+        ("low", "medium"),
+        {"low": 0, "medium": 0, "high": 25, "unknown": 33, "sum_of_filed_rules": 58}
+    ),
+    (
+        ("high", "unknown"),
+        {"low": 33, "medium": 351, "high": 0, "unknown": 0, "sum_of_filed_rules": 384}
+    ),
+    (
+        ("medium", "high"),
+        {"low": 33, "medium": 0, "high": 0, "unknown": 33, "sum_of_filed_rules": 66}
+    ),
+    (
+        ("low", "unknown"),
+        {"low": 0, "medium": 351, "high": 25, "unknown": 0, "sum_of_filed_rules": 376}
+    ),
+    (
+        ("low", "medium", "high"),
+        {"low": 0, "medium": 0, "high": 0, "unknown": 33, "sum_of_filed_rules": 33}
+    ),
+    (
+        ("low", "medium", "unknown"),
+        {"low": 0, "medium": 0, "high": 25, "unknown": 0, "sum_of_filed_rules": 25}
+    ),
+    (
+        ("low", "high", "unknown"),
+        {"low": 0, "medium": 351, "high": 0, "unknown": 0, "sum_of_filed_rules": 351}
+    ),
+    (
+        ("medium", "high", "unknown"),
+        {"low": 33, "medium": 0, "high": 0, "unknown": 0, "sum_of_filed_rules": 33}
+    ),
+])
+def test_report_severity_of_failed_rules_stats(to_remove, result):
+    report = None
+    if "fail" in to_remove:
```

Comment:
This is somehow suspicious condition.
I propose either split this test case into 2 (first with severity and second with test result) or add another parameter `type`.